pkg/catalog/manager/crd.go

```diff
@@ -166,6 +176,14 @@ func (m *Manager) createTemplateVersions(catalogName string, versionsSpec []v3.T
 		templateVersion.Labels = map[string]string{
 			TemplateNameLabel: template.Name,
 		}
+		//help with garbage collection on delete
+		ownerRef := []metav1.OwnerReference{{
+			Name:       catalogName,
+			APIVersion: "v3",
+			UID:        template.UID,
+			Kind:       template.Kind,
+		}}
```

Comment:
I think you either want the name to be the template name or the UID/Kind to be the catalog's UID/Kind, depending on which you want to be the owner.